NetTopologySuite.IO/NetTopologySuite.IO.PostGis/PostGisReader.cs

```diff
@@ -116,9 +110,7 @@ public IGeometry Read(Stream stream)
         /// <returns></returns>
         protected IGeometry Read(BinaryReader reader)
         {
-            // Dummy read, just for bytes compatibility.
-            // The byte order is determined only once.
-            reader.ReadByte();
+            ((BiEndianBinaryReader)reader).Endianess = (ByteOrder)reader.ReadByte();

```

Comment:
Won't this cast fail because we don't create it as `BiEndianBinaryReader` on line 97?

Reply:
Without changing line 97 to actually create it as `BiEndianBinaryReader`, that wouldn't actually help because it would just never set the property.

Reply:
My mistake. Fixed now. Do you have tests for geometry readers?